server/src/main/java/org/opensearch/repositories/RepositoryException.java

```diff
@@ -69,6 +70,15 @@ public RepositoryException(StreamInput in) throws IOException {
         repository = in.readOptionalString();
     }
 
+    @Override
+    public RestStatus status() {
+        String causeDescription = unwrapCause().getMessage();
+        if (causeDescription != null && causeDescription.contains("not found")) {
```

Comment:
This isn't very reliable. 

Start by writing a unit test that reproduces the problem, then we can talk about how to fix it best.

Reply:
make sense, I will try to write an integration/sociable test.